2/Dockerfile

```diff
@@ -0,0 +1,56 @@
+FROM centos:centos7
+
+# Jenkins image for OpenShift
+#
+# This image provides a Jenkins server, primarily intended for integration with
+# OpenShift v3.
+#
+# Volumes: 
+# * /var/jenkins_home
+# Environment:
+# * $JENKINS_PASSWORD - Password for the Jenkins 'admin' user.
+
+MAINTAINER Ben Parees <bparees@redhat.com>
+
+ENV JENKINS_VERSION=2.7 \
+    HOME=/var/lib/jenkins \
+    JENKINS_HOME=/var/lib/jenkins \
+    JENKINS_UC=https://updates.jenkins-ci.org
+
+LABEL k8s.io.description="Jenkins is a continuous integration server" \
+      k8s.io.display-name="Jenkins 2.7" \
+      openshift.io.expose-services="8080:http" \
+      openshift.io.tags="jenkins,jenkins1,ci" \
+      io.openshift.s2i.scripts-url=image:///usr/libexec/s2i
+
+# 8080 for main web interface, 50000 for slave agents
+EXPOSE 8080 50000
+
+RUN curl http://pkg.jenkins-ci.org/redhat-stable/jenkins.repo -o /etc/yum.repos.d/jenkins.repo && \
+    rpm --import http://pkg.jenkins-ci.org/redhat-stable/jenkins-ci.org.key && \
+    yum -y --setopt=tsflags=nodocs install epel-release && \
+    INSTALL_PKGS="rsync gettext git tar zip unzip java-1.8.0-openjdk jenkins-2.7.3-1.1 nss_wrapper" && \
+    yum -y --setopt=tsflags=nodocs install $INSTALL_PKGS && \
+    rpm -V $INSTALL_PKGS && \
+    yum clean all  && \
+    localedef -f UTF-8 -i en_US en_US.UTF-8 && \
+    curl -L https://github.com/openshift/origin/releases/download/v1.2.1/openshift-origin-client-tools-v1.2.1-5e723f6-linux-64bit.tar.gz | tar -zx && \
+    mv openshift*/oc /usr/local/bin && \
+    rm -rf openshift-origin-client-tools-*
+
+COPY ./contrib/openshift /opt/openshift
+COPY ./contrib/jenkins /usr/local/bin
+ADD ./contrib/s2i /usr/libexec/s2i
+
+RUN /usr/local/bin/install-plugins.sh /opt/openshift/base-plugins.txt && \
+    # need to create <plugin>.pinned files when upgrading "core" plugins like credentials or subversion that are bundled with the jenkins server
+    # Currently jenkins v2 does not embed any plugins, but for reference:
```

Comment:
OK, sure, it is still easy enough to discern what is up, I'm fine with it.

On Fri, Sep 9, 2016 at 1:25 PM, Ben Parees <notifications@github.com> wrote:

> In 2/Dockerfile
> <https://github.com/openshift/jenkins/pull/152#discussion_r78217103>:
>
> > +    INSTALL_PKGS="rsync gettext git tar zip unzip java-1.8.0-openjdk jenkins-2.7.3-1.1 nss_wrapper" && \
> > +    yum -y --setopt=tsflags=nodocs install $INSTALL_PKGS && \
> > +    rpm -V $INSTALL_PKGS && \
> > +    yum clean all  && \
> > +    localedef -f UTF-8 -i en_US en_US.UTF-8 && \
> > +    curl -L https://github.com/openshift/origin/releases/download/v1.2.1/openshift-origin-client-tools-v1.2.1-5e723f6-linux-64bit.tar.gz | tar -zx && \
> > +    mv openshift*/oc /usr/local/bin && \
> > +    rm -rf openshift-origin-client-tools-*
> > +
> > +COPY ./contrib/openshift /opt/openshift
> > +COPY ./contrib/jenkins /usr/local/bin
> > +ADD ./contrib/s2i /usr/libexec/s2i
> > +
> > +RUN /usr/local/bin/install-plugins.sh /opt/openshift/base-plugins.txt && \
> > +    # need to create <plugin>.pinned files when upgrading "core" plugins like credentials or subversion that are bundled with the jenkins server
> > +    # Currently jenkins v2 does not embed any plugins, but for reference:
>
> I did it this way so we can just delete line 47-48 if we need to add a pin
> line and be back to where we were.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/jenkins/pull/152/files/a9998960d38c7278ee3ee99104150ee6dd4ced37#r78217103>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADbadHs7s0jQFnqrnL4QoPgYjHZkcTH5ks5qoZZ2gaJpZM4J5LRY>
> .
>